packages/access-api/src/routes/validate-email.js

```diff
@@ -152,28 +153,30 @@ async function authorize(req, env) {
 
     if (confirmation.error) {
       throw new Error(`unable to validate access session: ${confirmation}`, {
-        cause: confirmation.error,
+        cause: confirmation,
       })
     }
 
-    const confirm = provide(
-      Access.confirm,
-      async ({ capability, invocation }) => {
+    const confirm = provideAdvanced({
+      capability: Access.confirm,
+      audience: literal(request.audience.did()),
+      handler: async ({ capability, invocation }) => {
         return accessConfirm.handleAccessConfirm(
           /** @type {Ucanto.Invocation<import('@web3-storage/access/types').AccessConfirm>} */ (
             invocation
           ),
           env
         )
-      }
-    )
+      },
+    })
     const confirmResult = await confirm(request, {
-      id: env.signer.verifier,
+      // note: this is required, but the validation will use `options.capability` passsed to `provideAdvanced`
+      id: env.signer,
```

Comment:
I find this comment confusing, maybe you can try rephrasing it. Specifically you talk about `options.capability` while you specify `env.signer` so it's not really clear.

Reply:
sorry I meant `options.audience`. Ended up moving away from this anyway and back to `provide` after we spoke f2f